Update package status

replaceafill opened the pull request:
This PR updates the package status when it's deleted from the file system and when it's requested to be deleted or recovered. This fixes the Status column in the Archival storage of the Dashboard and in the Packages table of the Storage Service and the Delete modal for DIPs.

Connected to https://github.com/archivematica/Issues/issues/1014

ross-spencer commented on storage_service/locations/tests/test_api.py:
```diff
@@ -521,6 +521,44 @@ def test_download_file_arkivum_error(self):
+    def test_recover_aip_request_updates_package_status(self):
+        self._test_request_view_updates_package_status(
+            "recover_aip", models.Package.RECOVER_REQ
+        )
```
Nice, I like the way you've laid this out for testing.

ross-spencer commented on storage_service/locations/tests/test_package.py:
```diff
@@ -74,12 +74,16 @@ def verify_redirect_message(response, message):
 
         # It returns a "success" message when the package was deleted
-        # successfully.
+        # successfully and updates its status
```
Good attention to detail.

ross-spencer reviewed (approved):
Really nice work again @replaceafill. Really great to read from and learn from. LGTM! :+1:

replaceafill committed:
Update package status

This updates the package status when it's deleted from the file
system and when it's requested to be deleted or recovered. This fixes
the Status column in the Archival storage of the Dashboard and in the
Packages table of the Storage Service and the Delete modal for DIPs.

replaceafill commented:
Thank you @ross-spencer!
